addons/account/models/account_move.py

```diff
@@ -4782,6 +4761,32 @@ def _message_post_after_hook(self, new_message, message_values):
 
         return res
 
+    def _check_and_decode_attachment(self, attachments):
+        if not attachments or self.env.context.get('no_new_invoice') or not self.is_invoice(include_receipts=True):
+            return False
+        if self.state != 'draft':
+            self.with_user(SUPERUSER_ID).message_post(
+                body=_('The invoice is not a draft, it was not updated from the attachment.'),
+                message_type='comment',
+            )
+            return False
+
+        # As we are coming from the mail, we assume that ONE of the attachments
+        # will enhance the invoice thanks to EDI / OCR / .. capabilities
+        decodable_attachment_per_inv = self._extend_with_attachments(attachments, new=bool(self._context.get('from_alias')))
```

Comment:
Don't you meant the opposite ? since it returns a mapping of the form `{ir.attachment(773,): account.move(31,)}`, I would name it `move_to_decodable_attachment` ?